server/services/client.ts

```diff
@@ -188,40 +187,48 @@ const clientService: (context: StrapiContext) => IClientService = ({ strapi }) =
     };
   },
 
-  renderTree(
+  async renderTree(
     items = [],
     id = null,
     field = 'parent',
     path = '',
     itemParser = (i: ToBeFixed) => i,
   ) {
-    return items
-      .filter(
-        (item) => {
-          if (item[field] === null && id === null) {
-            return true;
-          }
-          let data = item[field];
-          if (data && typeof id === 'string') {
-            data = data.toString();
-          }
-          if (!!data && typeof data === 'object' && 'id' in data) {
-            return data.id === id
-          }
-
-          return (data && data === id);
-        },
+    return (await Promise.all(
+      items
+        .filter(
+          (item) => {
+            if (item[field] === null && id === null) {
+              return true;
+            }
+            let data = item[field];
+            if (data && typeof id === 'string') {
+              data = data.toString();
+            }
+            if (!!data && typeof data === 'object' && 'id' in data) {
+              return data.id === id
+            }
+
+            return (data && data === id);
+          },
```

Comment:
extract this callback to separate function

Reply:
Most of the changes are from handling `async/await`. Point is valid, but outside of this story's scope.